node-it/src/test/scala/com/wavesplatform/it/sync/grpc/UpdateAssetInfoTransactionGrpcSuite.scala

```diff
@@ -0,0 +1,133 @@
+    val (defaultName, defaultDescription) = ("asset", "description")
+    assetId = Base58.encode(PBTransactions.vanilla(
+      sender.grpc.broadcastIssue(issuer, defaultName, someAssetAmount, 8, reissuable = true, script = None, fee = issueFee, description = defaultDescription, version = 1, waitForTx = true)
+    ).explicitGet().id())
+    issueHeight = sender.grpc.height
+  }
+
+  test("able to update name/description of issued asset") {
+    val nextTerm = issueHeight + updateInterval + 1
+    sender.grpc.waitForHeight(nextTerm)
+    val updateAssetInfoTxId = Base58.encode(PBTransactions.vanilla(
+      sender.grpc.updateAssetInfo(issuer, assetId, "updatedName", "updatedDescription", minFee)
+    ).explicitGet().id())
+    sender.grpc.waitForTransaction(updateAssetInfoTxId)
+
+    sender.grpc.getTransaction(assetId).getTransaction.getUpdateAssetInfo.name shouldBe "updatedName"
+    sender.grpc.getTransaction(assetId).getTransaction.getUpdateAssetInfo.description shouldBe "updatedDescription"
+    sender.grpc.assetInfo(assetId).name shouldBe "updatedName"
+    sender.grpc.assetInfo(assetId).description shouldBe "updatedDescription"
+  }
+
+  test("not able to update name/description more than once within interval") {
+    assertGrpcError(
+      sender.grpc.updateAssetInfo(issuer, assetId, "updatedName", "updatedDescription", minFee),
+      "Can't update asset info before",
+      Code.INVALID_ARGUMENT
+    )
+  }
+
+  val invalidAssetsNames =
+    Table(
+      "abc",
+      "UpperCaseAssetCoinTest",
+      "~!|#$%^&*()_+=\";:/?><|\\][{}"
+    )
+
+  forAll(invalidAssetsNames) { assetName: String =>
+    test(s"not able to update name to $assetName") {
+      sender.grpc.waitForHeight(sender.height + 3)
+      assertGrpcError(
+        sender.grpc.updateAssetInfo(issuer, assetId, assetName, "updatedDescription", minFee),
+        "invalid name",
+        Code.INVALID_ARGUMENT
+      )
+    }
+  }
+
+  test(s"not able to set too big description") {
+    val tooBigDescription = Random.nextString(1001)
+    assertGrpcError(
+      sender.grpc.updateAssetInfo(issuer, assetId, "updatedName", tooBigDescription, minFee),
+      "Too big sequences requested",
+        Code.INVALID_ARGUMENT
+    )
+  }
+
+  test("not able to update asset info without paying enough fee") {
+    assertGrpcError(
+      sender.updateAssetInfo(issuer, assetId, "updatedName", "updatedDescription", minFee - 1),
+      s"Fee for UpdateAssetInfoTransaction (${minFee - 1} in WAVES) does not exceed minimal value of $minFee WAVES.",
+      Code.INVALID_ARGUMENT
+    )
+  }
+
+  test("not able to update info of not-issued asset") {
+    val notIssuedAssetId = "BzARFPgBqWFu6MHGxwkPVKmaYAzyShu495Ehsgru72Wz"
+    assertGrpcError(
+      sender.updateAssetInfo(issuer, notIssuedAssetId, "updatedName", "updatedDescription", minFee),
+      "Referenced assetId not found",
+      Code.INVALID_ARGUMENT
+    )
+  }
+
+  test("non-issuer cannot update asset info") {
```

Comment:
Is there unit tests for smart sender/token extra fees?

Reply:
There is no unit test for it, but here is an integration test
https://github.com/wavesplatform/Waves/pull/2871/files#diff-04882afc75a1259824fc515be37793dcR145